File: examples/ciflyRExamples/R/minDsep.R
library("here")
source(here("R", "nearestDsep.R"))

findMinSeparator <- function(g, X, Y, I, R) {
  Zx <- findNearestSeparator(g, X, Y, I, R)
  if (is.null(Zx)) {
    return (NULL)
  }
  Zy <- findNearestSeparator(g, Y, X, I, Zx)
  if (is.null(Zy)) {
    return (NULL)
  }
  return(union(intersect(Zx, Zy), I))
}

File: examples/ciflypy-examples/ciflypy_examples/min_dsep.py
from ciflypy_examples.nearest_dsep import find_nearest_separator

def find_min_separator(g, X, Y, I, R):
    Zx = find_nearest_separator(g, X, Y, I, R)
    if not Zx:
        return None
    Zy = find_nearest_separator(g, Y, X, I, Zx)
    if not Zy:
        return None
    return set(Zx).intersection(Zy).union(I)
